### DISCUSSION TITLE
Whitelist named constructor via annotation

### DISCUSSION THREAD

**Discussion by magnetik** [CuyZ/Valinor#99] [Ideas] [answered]
Regarding the 0.7 release and the security advisory, I was looking for a way to *not* have to define the constructor for every call.

Could some kind of "constructor whitelist" through an attribute can be implemented?

Thanks again for your work on the lib :)

**Comment by romm** [comment]
Hi @magnetik, this was already requested by someone else and is probably a relevant feature to add. I'll be thinking about adding it to the library and keep you updated here.

**Reply by magnetik** [reply]
Any news for this? Since Valinor usage have increased a lot maybe some others came with this need?

**Comment by romm** [comment]
Hi @magnetik here's a present for you: #484

Would you mind taking a look at it? 🙂

**Reply by magnetik** [reply]
I somehow missed all notifications about this thread and just bumped into it randomly. 

It looks like it fits perfectly. Thanks.

**Comment by romm** [comment, marked as answer]
Support for `Constructor` attribute has been added in [v1.10.0](https://github.com/CuyZ/Valinor/releases/tag/1.10.0)!

**Reply by magnetik** [reply]
awesome ! Thanks.
